### PR TITLE
[LabKey Cloud] issue 49843: update ciphers

### PR DESCRIPTION
**Pull request by labkey-willm**
#### Rationale
https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=49843

default ciphers were out of date for tomcat 10. Updated per: https://tomcat.apache.org/tomcat-10.1-doc/config/http.html#SSL_Support_-_SSLHostConfig

also, latest docker-compose was resolving all the IDENT vars in the docker-compose.yml at once, so it thought there were conflicts. Had to uniquify.

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

**Review by labkey-stuartm — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -73,9 +73,9 @@ ENV DEBUG="${DEBUG}" \
     TOMCAT_KEYSTORE_FORMAT="PKCS12" \
     TOMCAT_KEYSTORE_ALIAS="tomcat" \
     \
-    TOMCAT_SSL_CIPHERS="HIGH:!ADH:!EXP:!SSLv2:!SSLv3:!MEDIUM:!LOW:!NULL:!aNULL" \
+    TOMCAT_SSL_CIPHERS="HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!kRSA" \
```
Comment (labkey-jeckels) [inline Dockerfile]:
```suggestion
    TOMCAT_SSL_CIPHERS="HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!kRSA:!EDH:!DHE:!DH:!ECDH" \
```

File: application.properties [hunk 2]
```diff
@@ -121,8 +121,8 @@ server.http2.enabled=true
 
 server.ssl.enabled=true
 
-server.ssl.ciphers=${TOMCAT_SSL_CIPHERS:-HIGH:!ADH:!EXP:!SSLv2:!SSLv3:!MEDIUM:!LOW:!NULL:!aNULL}
-server.ssl.enabled-protocols=${TOMCAT_SSL_ENABLED_PROTOCOLS:-TLSv1.3,+TLSv1.2}
+server.ssl.ciphers=${TOMCAT_SSL_CIPHERS:-HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!kRSA}
```
Comment (labkey-jeckels) [inline application.properties]:
```suggestion
server.ssl.ciphers=${TOMCAT_SSL_CIPHERS:-HIGH:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!kRSA:!EDH:!DHE:!DH:!ECDH}
```

File: application.properties [hunk 3]
```diff
@@ -121,8 +121,8 @@ server.http2.enabled=true
...
+server.ssl.enabled-protocols=${TOMCAT_SSL_ENABLED_PROTOCOLS:-TLSv1.3,TLSv1.2}
```
Comment (labkey-jeckels) [inline application.properties:125]: The doesn't match the value above, and it breaks the server when I try it. I also think we want TLS 1.3 enabled.

```suggestion
server.ssl.enabled-protocols=${TOMCAT_SSL_ENABLED_PROTOCOLS:TLSv1.3,TLSv1.2}
```

File: docker-compose.yml [hunk 4]
```diff
@@ -361,7 +361,7 @@ services:
       - LOG_LEVEL_SPRING_WEB=INFO
       - LOG_LEVEL_TOMCAT=DEBUG
       - LOG_LEVEL_DEFAULT=DEBUG
-      - LOG_LEVEL_LABKEY_DEFAULT=DEBUG
+      - LOG_LEVEL_LABKEY_DEFAULT=INFO
```
Comment (labkey-jeckels) [inline docker-compose.yml]: Is this intentional? It's not mentioned in the PR description, so I wanted to double-check

**Review by labkey-jeckels — APPROVED** [review]
I'll capture more detailed feedback in the issue, but I can't see a way to get an A rating from SSL Labs without getting a warning from Tomcat. I hate the spam in the log, but if I have to choose, I'd rather improve security than suppress the log message.

File: docker-compose.yml (hunk 4 above)
Comment (labkey-willm) [reply docker-compose.yml]: nope. Must have been leftover from testing. I'll revert. Thanks.

File: application.properties (hunk 3 above)
Comment (labkey-willm) [reply application.properties:125]: I removed the + on purpose, as it looked like a long-standing typo.

the - in that context is supposed to tell it to use what follows as a default if the var isn't defined. How did it break yours?

Comment (labkey-jeckels) [reply application.properties:125]: Ah, OK. I misunderstood the syntax. I tried setting my `application.properties` to use `server.ssl.enabled-protocols=-TLSv1.3,TLSv1.2` which broke things, not realizing that the `-` wasn't part of the value that got passed through.

**Pull request merged**